### ISSUE TITLE
GKE Workflow pending on running phase (multiple argo versions)

### ISSUE THREAD

**Issue by thiago4int** [argoproj/argo-workflows#5967]
## Summary 

Even building a new gke cluster from scratch and trying multiple argo versions the workflow keeps hanging without a ready status.

## Diagnostics

1.19.9-gke.1400
What version of Argo Workflows are you running? 
EDIT: 3.1, 3.0.4, 3.0.3, 3.0.2, 2.12 and 2.12.9.

```
apiVersion: v1
kind: Pod
metadata:
  annotations:
    workflows.argoproj.io/node-name: hello-world-nb7bl
    workflows.argoproj.io/template: '{"name":"whalesay","inputs":{},"outputs":{},"metadata":{},"container":{"name":"","image":"docker/whalesay:latest","command":["cowsay"],"args":["hello
      world"],"resources":{}}}'
  creationTimestamp: "2021-05-21T15:48:58Z"
  labels:
    workflows.argoproj.io/completed: "false"
    workflows.argoproj.io/workflow: hello-world-nb7bl
  managedFields:
  - apiVersion: v1
    fieldsType: FieldsV1
    fieldsV1:
      f:metadata:
        f:annotations:
          .: {}
          f:workflows.argoproj.io/node-name: {}
          f:workflows.argoproj.io/template: {}
        f:labels:
          .: {}
          f:workflows.argoproj.io/completed: {}
          f:workflows.argoproj.io/workflow: {}
        f:ownerReferences:
          .: {}
          k:{"uid":"38373f01-0148-46c2-85d2-ac45af92197c"}:
            .: {}
            f:apiVersion: {}
            f:blockOwnerDeletion: {}
            f:controller: {}
            f:kind: {}
            f:name: {}
            f:uid: {}
      f:spec:
        f:containers:
          k:{"name":"main"}:
            .: {}
            f:args: {}
            f:command: {}
            f:env:
              .: {}
              k:{"name":"ARGO_CONTAINER_NAME"}:
                .: {}
                f:name: {}
                f:value: {}
              k:{"name":"ARGO_INCLUDE_SCRIPT_OUTPUT"}:
                .: {}
                f:name: {}
                f:value: {}
            f:image: {}
            f:imagePullPolicy: {}
            f:name: {}
            f:resources: {}
            f:terminationMessagePath: {}
            f:terminationMessagePolicy: {}
          k:{"name":"wait"}:
            .: {}
            f:command: {}
            f:env:
              .: {}
              k:{"name":"ARGO_CONTAINER_NAME"}:
                .: {}
                f:name: {}
                f:value: {}
              k:{"name":"ARGO_CONTAINER_RUNTIME_EXECUTOR"}:
                .: {}
                f:name: {}
              k:{"name":"ARGO_INCLUDE_SCRIPT_OUTPUT"}:
                .: {}
                f:name: {}
                f:value: {}
              k:{"name":"ARGO_POD_NAME"}:
                .: {}
                f:name: {}
                f:valueFrom:
                  .: {}
                  f:fieldRef:
                    .: {}
                    f:apiVersion: {}
                    f:fieldPath: {}
              k:{"name":"GODEBUG"}:
                .: {}
                f:name: {}
                f:value: {}
            f:image: {}
            f:imagePullPolicy: {}
            f:name: {}
            f:resources: {}
            f:terminationMessagePath: {}
            f:terminationMessagePolicy: {}
            f:volumeMounts:
              .: {}
              k:{"mountPath":"/argo/podmetadata"}:
                .: {}
                f:mountPath: {}
                f:name: {}
              k:{"mountPath":"/var/run/docker.sock"}:
                .: {}
                f:mountPath: {}
                f:name: {}
                f:readOnly: {}
        f:dnsPolicy: {}
        f:enableServiceLinks: {}
        f:restartPolicy: {}
        f:schedulerName: {}
        f:securityContext: {}
        f:terminationGracePeriodSeconds: {}
        f:volumes:
          .: {}
          k:{"name":"docker-sock"}:
            .: {}
            f:hostPath:
              .: {}
              f:path: {}
              f:type: {}
            f:name: {}
          k:{"name":"podmetadata"}:
            .: {}
            f:downwardAPI:
              .: {}
              f:defaultMode: {}
              f:items: {}
            f:name: {}
    manager: workflow-controller
    operation: Update
    time: "2021-05-21T15:48:58Z"
  - apiVersion: v1
    fieldsType: FieldsV1
    fieldsV1:
      f:status:
        f:conditions:
          k:{"type":"ContainersReady"}:
            .: {}
            f:lastProbeTime: {}
            f:lastTransitionTime: {}
            f:message: {}
            f:reason: {}
            f:status: {}
            f:type: {}
          k:{"type":"Initialized"}:
            .: {}
            f:lastProbeTime: {}
            f:lastTransitionTime: {}
            f:status: {}
            f:type: {}
          k:{"type":"Ready"}:
            .: {}
            f:lastProbeTime: {}
            f:lastTransitionTime: {}
            f:message: {}
            f:reason: {}
            f:status: {}
            f:type: {}
        f:containerStatuses: {}
        f:hostIP: {}
        f:phase: {}
        f:podIP: {}
        f:podIPs:
          .: {}
          k:{"ip":"10.80.2.16"}:
            .: {}
            f:ip: {}
        f:startTime: {}
    manager: kubelet
    operation: Update
    time: "2021-05-21T15:49:03Z"
  name: hello-world-nb7bl
  namespace: argo
  ownerReferences:
  - apiVersion: argoproj.io/v1alpha1
    blockOwnerDeletion: true
    controller: true
    kind: Workflow
    name: hello-world-nb7bl
    uid: 38373f01-0148-46c2-85d2-ac45af92197c
  resourceVersion: "44261"
  selfLink: /api/v1/namespaces/argo/pods/hello-world-nb7bl
  uid: 46dfe5f9-282c-49ce-ade5-e2be37510398
spec:
  containers:
  - command:
    - argoexec
    - wait
    - --loglevel
    - info
    env:
    - name: ARGO_POD_NAME
      valueFrom:
        fieldRef:
          apiVersion: v1
          fieldPath: metadata.name
    - name: ARGO_CONTAINER_RUNTIME_EXECUTOR
    - name: GODEBUG
      value: x509ignoreCN=0
    - name: ARGO_CONTAINER_NAME
      value: wait
    - name: ARGO_INCLUDE_SCRIPT_OUTPUT
      value: "false"
    image: argoproj/argoexec:v3.1.0-rc5
    imagePullPolicy: IfNotPresent
    name: wait
    resources: {}
    terminationMessagePath: /dev/termination-log
    terminationMessagePolicy: File
    volumeMounts:
    - mountPath: /argo/podmetadata
      name: podmetadata
    - mountPath: /var/run/docker.sock
      name: docker-sock
      readOnly: true
    - mountPath: /var/run/secrets/kubernetes.io/serviceaccount
      name: default-token-4mqc5
      readOnly: true
  - args:
    - hello world
    command:
    - cowsay
    env:
    - name: ARGO_CONTAINER_NAME
      value: main
    - name: ARGO_INCLUDE_SCRIPT_OUTPUT
      value: "false"
    image: docker/whalesay:latest
    imagePullPolicy: Always
    name: main
    resources: {}
    terminationMessagePath: /dev/termination-log
    terminationMessagePolicy: File
    volumeMounts:
    - mountPath: /var/run/secrets/kubernetes.io/serviceaccount
      name: default-token-4mqc5
      readOnly: true
  dnsPolicy: ClusterFirst
  enableServiceLinks: true
  nodeName: gke-argo-events-stage-default-pool-f489baa5-qfzn
  preemptionPolicy: PreemptLowerPriority
  priority: 0
  restartPolicy: Never
  schedulerName: default-scheduler
  securityContext: {}
  serviceAccount: default
  serviceAccountName: default
  terminationGracePeriodSeconds: 30
  tolerations:
  - effect: NoExecute
    key: node.kubernetes.io/not-ready
    operator: Exists
    tolerationSeconds: 300
  - effect: NoExecute
    key: node.kubernetes.io/unreachable
    operator: Exists
    tolerationSeconds: 300
  volumes:
  - downwardAPI:
      defaultMode: 420
      items:
      - fieldRef:
          apiVersion: v1
          fieldPath: metadata.annotations
        path: annotations
    name: podmetadata
  - hostPath:
      path: /var/run/docker.sock
      type: Socket
    name: docker-sock
  - name: default-token-4mqc5
    secret:
      defaultMode: 420
      secretName: default-token-4mqc5
status:
  conditions:
  - lastProbeTime: null
    lastTransitionTime: "2021-05-21T15:48:58Z"
    status: "True"
    type: Initialized
  - lastProbeTime: null
    lastTransitionTime: "2021-05-21T15:48:58Z"
    message: 'containers with unready status: [main]'
    reason: ContainersNotReady
    status: "False"
    type: Ready
  - lastProbeTime: null
    lastTransitionTime: "2021-05-21T15:48:58Z"
    message: 'containers with unready status: [main]'
    reason: ContainersNotReady
    status: "False"
    type: ContainersReady
  - lastProbeTime: null
    lastTransitionTime: "2021-05-21T15:48:58Z"
    status: "True"
    type: PodScheduled
  containerStatuses:
  - containerID: containerd://e36a598b11d46b1418db501d14a79f1b488ca4c2a56b0aa0ac1e0b811b69d550
    image: docker.io/docker/whalesay:latest
    imageID: sha256:c717279bbba020bf95ac72cf47b2c8abb3a383ad4b6996c1a7a9f2a7aaa480ad
    lastState: {}
    name: main
    ready: false
    restartCount: 0
    started: false
    state:
      terminated:
        containerID: containerd://e36a598b11d46b1418db501d14a79f1b488ca4c2a56b0aa0ac1e0b811b69d550
        exitCode: 0
        finishedAt: "2021-05-21T15:49:02Z"
        reason: Completed
        startedAt: "2021-05-21T15:49:02Z"
  - containerID: containerd://2f45f848ae4a50b26b6c240347986880d7e61d5dfb8947d26e67e81428cbb59b
    image: docker.io/argoproj/argoexec:v3.1.0-rc5
    imageID: docker.io/argoproj/argoexec@sha256:30f84f8f8207db460d9d67b596b82310619f727cfbe1d5b24de8c55265beb0f5
    lastState: {}
    name: wait
    ready: true
    restartCount: 0
    started: true
    state:
      running:
        startedAt: "2021-05-21T15:49:01Z"
  hostIP: 10.128.15.217
  phase: Running
  podIP: 10.80.2.16
  podIPs:
  - ip: 10.80.2.16
  qosClass: BestEffort
  startTime: "2021-05-21T15:48:58Z"

   
  
```

https://drive.google.com/file/d/19GbrhuPtnJtg-Rk6gvJ5tqu9vI3VTrIy/view?usp=sharing

---

**Message from the maintainers**:

Impacted by this bug? Give it a 👍. We prioritise the issues with the most 👍.

**Comment by thiago4int** [comment]
Not working on v3.1.0-rc6 as well

**Comment by alexec** [comment]
Can you try v2.12?

**Comment by thiago4int** [comment]
@alexec 
Changed the error. Now I have this output: 
https://drive.google.com/file/d/1kG3rfdxmvbcyA9zo4YQueB0szpV4j3Zo/view?usp=sharing

**Comment by alexec** [comment]
OK. So issue with GKE. Can I ask you try a different executor, e.g. `k8sapi` or `pns`?

**Comment by thiago4int** [comment]
@alexec
Sure, I will try to reproduce the same on a different environment. Is there any possible workaround? 

@caueasantos 
Can you tell me which version of argo you run on GKE and which version and image of the nodes?

**Comment by thiago4int** [comment]
@alexec 
Is there any docs or even a template in order to change the executor on the configmap? The most intriguing part is that nothing was changed before this error became permanent.

**Comment by thiago4int** [comment]
@alexec 
Workaround:
- downgrade 1.18.17-gke.100
- Ubuntu with Docker as the container runtime
- Argo 3.0.2

Works for me

**Comment by caueasantos** [comment]
> @alexec
> Sure, I will try to reproduce the same on a different environment. Is there any possible workaround?
> 
> @caueasantos
> Can you tell me which version of argo you run on GKE and which version and image of the nodes?

Sure. 

```shell
GKE 1.19.9-gke.1900
Argo v3.0.3 
```

**Comment by thiago4int** [comment]
@caueasantos 
What about your container runtime? Have you ever tried Container Optimized OS?

**Comment by caueasantos** [comment]
> @caueasantos
> What about your container runtime? Have you ever tried Container Optimized OS?

I've been using containerd for a long time

```yaml
OS-IMAGE: Container-Optimized OS from Google
KERNEL-VERSION: 5.4.89+
CONTAINER-RUNTIME: containerd://1.4.3
```

**Comment by alexec** [comment]
Marking as invalid as this can be fixed by changing executor.

**Comment by alexec** [comment]
I think this is probably:

Duplicate of $6256
